### PR TITLE
Add tests for project vertex shader functions

### PR DESCRIPTION
**Pull request by Pessimistress**
For https://github.com/uber/deck.gl/issues/2272

#### Change List
- Add tests for `project` module's vertex shader functions

#### TODO
- More tests for modelMatrix will be added in follow up PRs as I address the CPU/GPU disparity issue
- Use `Transform` instead of transpilation in browser test @1chandu

### CODE REVIEW

File: test/modules/core/shaderlib/shaderlib-test-utils.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+});
+
+// @returns JavaScript function of the transpiled shader
+export function compileVertexShader(source) {
+  source = normalize(source);
+
+  const compiledSource = compileVS(source);
+  const {compiler} = compileVS;
+
+  const stats = {
```
Comment (ibgreen) [inline test/modules/core/shaderlib/shaderlib-test-utils.js]: Nit: why the name `stats`? `programElements`... etc?

File: test/modules/core/shaderlib/shaderlib-test-utils.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// TODO - move to @luma.gl/debug ?
```
Comment (ibgreen) [inline test/modules/core/shaderlib/shaderlib-test-utils.js]: Or `luma.gl/shadertools`

File: test/modules/core/shaderlib/shaderlib-test-utils.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  return {
+    ${Object.keys(stats.functions).join(',')}
+  };
+}`
+  );
+}
+
+/* eslint-disable no-eval */
+function evalScript(value) {
```
Comment (ibgreen) [inline test/modules/core/shaderlib/shaderlib-test-utils.js]: Can `value` be better named? `scriptSource` etc...

File: test/modules/core/shaderlib/shaderlib-test-utils.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// TODO - move to @luma.gl/debug ?
+import Compiler from 'glsl-transpiler';
+
+const normalize = source =>
+  source
+    // prepr does not like #define without value
```
Comment (ibgreen) [inline test/modules/core/shaderlib/shaderlib-test-utils.js]: prepr -> GLSL preprocessor?

File: test/modules/core/shaderlib/shaderlib-test-utils.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// TODO - move to @luma.gl/debug ?
+import Compiler from 'glsl-transpiler';
+
+const normalize = source =>
```
Comment (ibgreen) [inline test/modules/core/shaderlib/shaderlib-test-utils.js]: Nit: `normalize` => `normalizeShaderSource`?

**Comment by 1chandu** [comment]
Yeah, we can use `Transform` to test on GPU, I haven't started these tests for `project` module yet, plan to add them soon.

**Pull request merged**